wav_steganography/wav_file.py

```diff
@@ -121,32 +126,62 @@ def plot(self, from_s: float = 0.0, to_s: Optional[float] = 0.1, filename: Union
 
     def encode(
             self,
-            message: bytes,
+            data: bytes,
             least_significant_bits: int = 2,
             every_nth_byte: int = 1,
-            password: Optional[str] = None
-    ):
+            redundant_bits: int = 4,
+            encryption_type: Optional[EncryptionType] = EncryptionType.NONE,
+            error_correction: bool = False
+        ):
         """ Encode a message in the given WAVFile
 
         This is done by writing to every nth bytes some number of least significant bits.
         A short header is written first, then the message.
         """
         assert least_significant_bits <= self.header["BitsPerSample"]
-        message_encoder = Message.Encoder(least_significant_bits, every_nth_byte, password)
+
+        message = Message()
+
+        encryptor = EncryptionProvider.get_encryptor(encryption_type)
+        encryptor.configure()
+
+        message.encode_message(
+            data,
+            least_significant_bits,
+            every_nth_byte,
+            redundant_bits,
+            encryptor,
+            error_correction)
+
         byte_index = 0
-        header, data = message_encoder.encode(message)
-        for chunk in [header, data]:
+        bits_required_for_data = len(message.data.data) * 8
+        bits_available = (len(self.data) - len(message.header.data) * 8) * least_significant_bits // every_nth_byte
+        if bits_available < bits_required_for_data:
+            raise ValueError(
```

Comment:
Ich habe das in einen ValueError umgebaut und einen Test hinzugefügt, der abprüft dass der Error auch kommt 😀